sql-server-2014-alwayson-existing-vnet-and-ad/azuredeploy.json

```diff
@@ -0,0 +1,531 @@
+    "deploySqlClusterVms": "deploySqlClusterVms",
+    "deploySqlClusterVmsId": "[concat('Microsoft.Resources/deployments/', variables('deploySqlClusterVms'))]",
+
+    "sqlWitnessTemplateURL": "[concat(parameters('assetLocation'),'/SqlWitnessVmTemplate.json')]",
+    "sqlWitnessStorageName": "[concat(parameters('newStorageAccountNamePrefix'),'sqlwsa')]",
+    "deploySqlWitnessVms": "deploySqlWitnessVms",
+    "deploySqlWitnessVmsId": "[concat('Microsoft.Resources/deployments/', variables('deploySqlWitnessVms'))]",
+
+    "updateSQLWNic": "updateSQLWNic",
+    "updateSQLWNicId": "[concat('Microsoft.Resources/deployments/', variables('updateSQLWNic'))]",
+    "sqlwNicName": "SqlWitnessMachine-Nic-0",
+    "nicTemplateURL": "[concat(parameters('assetLocation'),'/nic.json')]",
+
+    "updateSQLCNic1": "updateSQLCNic1",
+    "updateSQLCNic1Id": "[concat('Microsoft.Resources/deployments/', variables('updateSQLCNic1'))]",
+    "sqlNicName1": "SqlMachine-Nic-0",
+
+    "updateSQLCNic2": "updateSQLCNic2",
+    "updateSQLCNic2Id": "[concat('Microsoft.Resources/deployments/', variables('updateSQLCNic2'))]",
+    "sqlNicName2": "SqlMachine-Nic-1",
+
+    "publicLBName": "sqlLoadBalancer",
+    "lbFE": "SqlLBFE",
+    "lbBE": "SqlLBBE",
+    "sqllbID":"[resourceId('Microsoft.Network/loadBalancers',variables('publicLBName'))]",
+    "sqllbFEConfigID": "[concat(variables('sqllbID'),'/frontendIPConfigurations/',variables('lbFE'))]",
+    "sqlBEAddressPoolID": "[concat(variables('sqllbID'),'/backendAddressPools/',variables('lbBE'))]",
+
+    "deploySqlWitnessShare": "deploySqlWitnessShare",
+    "deploySqlWitnessShareId": "[concat('Microsoft.Resources/deployments/', variables('deploySqlWitnessShare'))]",
+    "deploySqlWitnessShareTemplateUrl": "[concat(parameters('assetLocation'),'/DeploySqlWitnessShare.json')]",
+    "sqlWitnessSharePath" : "[concat(parameters('vmNamePrefix'),'-fsw')]",
+    "sqlWitnessVmNamePrefix": "[concat(parameters('vmNamePrefix'),'-SqlW-')]",
+    "sqlwVMName": "[concat(variables('sqlWitnessVmNamePrefix'),'0')]",
+
+    "deploySqlAlwaysOn": "deploySqlAlwaysOn",
+    "deploySqlAlwaysOnTemplateUrl": "[concat(parameters('assetLocation'),'/DeploySqlAlwaysOn.json')]",
+
+    "sqlClusterVmNamePrefix": "[concat(parameters('vmNamePrefix'),'-SqlC-')]",
+    "sqlCVMName1": "[concat(variables('sqlClusterVmNamePrefix'),'0')]",
+    "sqlCVMName2": "[concat(variables('sqlClusterVmNamePrefix'),'0')]",
+
+    "sqlLBName": "sqlLoadBalancer",
+
+    "sqlAOEPName": "[concat(parameters('vmNamePrefix'),'-hadr')]",
+    "sqlAOAGName": "[concat(parameters('vmNamePrefix'),'-ag')]",
+    "sqlAOListenerName": "[concat(parameters('vmNamePrefix'),'ag-listener')]",
+
+    "sqlAvailabilitySetName": "sqlAvailabilitySet"
+  },
+
+  "resources": [
+
+    {
+      "type": "Microsoft.Compute/availabilitySets",
+      "name": "[variables('sqlAvailabilitySetName')]",
+      "apiVersion": "2015-05-01-preview",
```

Comment:
Please move to 2015-06-15 api version and also make apiVersion a variable that is reused for resources